File: 0501-find-mode-in-binary-search-tree/0501-find-mode-in-binary-search-tree.java
class Solution {
    public int[] findMode(TreeNode root) {
        if(root == null){
            return new int[0];
        }

        Map<Integer, Integer> freq = new HashMap<>();
        inorder(root, freq);

        int maxFreq = 0;
        for(int val : freq.values()){
            maxFreq = Math.max(maxFreq, val);
        }

        List<Integer> res = new ArrayList<>();
        for(Map.Entry<Integer, Integer> e : freq.entrySet()){
            if(e.getValue() == maxFreq){
                res.add(e.getKey());
            }
        }

        int[] ans = new int[res.size()];
        for(int i = 0; i < res.size(); i++){
            ans[i] = res.get(i);
        }
        return ans;
    }

    public void inorder(TreeNode root, Map<Integer,Integer> freq){
        if(root == null) return;

        inorder(root.left, freq);
        freq.put(root.val, freq.getOrDefault(root.val, 0) + 1);
        inorder(root.right, freq);
    }
}

File: 0501-find-mode-in-binary-search-tree/0501-find-mode-in-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findMode(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        
        hashmap = {}
        self.inorder(root, hashmap)
        
        max_freq = max(hashmap.values())
        
        # Collect all values with max frequency
        return [key for key, freq in hashmap.items() if freq == max_freq]

    def inorder(self, node, hashmap):
        if not node:
            return
        
        self.inorder(node.left, hashmap)
        
        hashmap[node.val] = hashmap.get(node.val, 0) + 1
        
        self.inorder(node.right, hashmap)
     
    
        